# ATURAN PEMBUATAN TUGAS

# Anggota Kelompok
1. 10122411- SAHL HAFIDZ IRKHAMI
2. Andre Mahesa Somantri-10122405
3. Wawi Rhiksa Wibawa-10122470

# Soal No 1

Apakah Yang Dimaksud dengan duplikat data pada bahasa pemrograman python

In [ ]:
"Duplikat data" dalam konteks bahasa pemrograman Python dapat merujuk pada situasi di mana terdapat nilai atau elemen yang sama yang muncul lebih dari satu kali dalam suatu struktur data. Struktur data seperti daftar (list), set, atau kamus (dictionary) dapat mengandung duplikat data."

# Soal No 2

In [2]:
import pandas as pd
file_path = 'Indian-Liver-Patient-Dataset-(ILPD).csv'
pd.read_csv(file_path)

65 Female   0.7  0.1  187  16   18  6.8  3.3   0.9  1
0    62   Male  10.9  5.5  699  64  100  7.5  3.2  0.74  1
1    62   Male   7.3  4.1  490  60   68  7.0  3.3  0.89  1
2    58   Male   1.0  0.4  182  14   20  6.8  3.4  1.00  1
3    72   Male   3.9  2.0  195  27   59  7.3  2.4  0.40  1
4    46   Male   1.8  0.7  208  19   14  7.6  4.4  1.30  1
..   ..    ...   ...  ...  ...  ..  ...  ...  ...   ... ..
577  60   Male   0.5  0.1  500  20   34  5.9  1.6  0.37  2
578  40   Male   0.6  0.1   98  35   31  6.0  3.2  1.10  1
579  52   Male   0.8  0.2  245  48   49  6.4  3.2  1.00  1
580  31   Male   1.3  0.5  184  29   32  6.8  3.4  1.00  1
581  38   Male   1.0  0.3  216  21   24  7.3  4.4  1.50  2

[582 rows x 11 columns]

# Soal No 3

In [4]:
import pandas as pd

file_path = 'Indian-Liver-Patient-Dataset-(ILPD).csv'

dataset = pd.read_csv(file_path)

duplicate_rows = dataset[dataset.duplicated()]

print("Duplikat Data:")
print(duplicate_rows)


Duplikat Data:
     65  Female  0.7  0.1  187   16   18  6.8  3.3   0.9  1
18   40  Female  0.9  0.3  293  232  245  6.8  3.1  0.80  1
25   34    Male  4.1  2.0  289  875  731  5.0  2.7  1.10  1
33   38  Female  2.6  1.2  410   59   57  5.6  3.0  0.80  2
54   42    Male  8.9  4.5  272   31   61  5.8  2.0  0.50  1
61   58    Male  1.0  0.5  158   37   43  7.2  3.6  1.00  1
105  36    Male  5.3  2.3  145   32   92  5.1  2.6  1.00  2
107  36    Male  0.8  0.2  158   29   39  6.0  2.2  0.50  2
137  18    Male  0.8  0.2  282   72  140  5.5  2.5  0.80  1
142  30    Male  1.6  0.4  332   84  139  5.6  2.7  0.90  1
157  72    Male  0.7  0.1  196   20   35  5.8  2.0  0.50  1
163  39    Male  1.9  0.9  180   42   62  7.4  4.3  1.38  1
173  31    Male  0.6  0.1  175   48   34  6.0  3.7  1.60  1
200  49    Male  0.6  0.1  218   50   53  5.0  2.4  0.90  1


# Soal No 4

Tuliskan Script untuk mencari duplikat data pada dataset Indian-Liver-Patient-Dataset-(ILPD).csv berdasarkan kolom 3
(ex data : 0.7, 10.9, 7.3)

In [6]:
import pandas as pd

file_path = 'Indian-Liver-Patient-Dataset-(ILPD).csv'

dataset = pd.read_csv(file_path, header=None)

column_index_to_display = 2
column_data = dataset.iloc[:, column_index_to_display]

print(f"Isi dari kolom ke-{column_index_to_display + 1}:")
print(column_data)


Isi dari kolom ke-3:
0       0.7
1      10.9
2       7.3
3       1.0
4       3.9
       ... 
578     0.5
579     0.6
580     0.8
581     1.3
582     1.0
Name: 2, Length: 583, dtype: float64


# Soal No 5

In [9]:
import pandas as pd

file_path = 'Indian-Liver-Patient-Dataset-(ILPD).csv'

dataset = pd.read_csv(file_path)

print("Duplikat Data Sebelum Dihapus:")
print(dataset[dataset.duplicated()])

dataset_no_duplicates = dataset.drop_duplicates()

print("\nDuplikat Data Setelah Dihapus:")
print(dataset_no_duplicates)

Duplikat Data Sebelum Dihapus:
     65  Female  0.7  0.1  187   16   18  6.8  3.3   0.9  1
18   40  Female  0.9  0.3  293  232  245  6.8  3.1  0.80  1
25   34    Male  4.1  2.0  289  875  731  5.0  2.7  1.10  1
33   38  Female  2.6  1.2  410   59   57  5.6  3.0  0.80  2
54   42    Male  8.9  4.5  272   31   61  5.8  2.0  0.50  1
61   58    Male  1.0  0.5  158   37   43  7.2  3.6  1.00  1
105  36    Male  5.3  2.3  145   32   92  5.1  2.6  1.00  2
107  36    Male  0.8  0.2  158   29   39  6.0  2.2  0.50  2
137  18    Male  0.8  0.2  282   72  140  5.5  2.5  0.80  1
142  30    Male  1.6  0.4  332   84  139  5.6  2.7  0.90  1
157  72    Male  0.7  0.1  196   20   35  5.8  2.0  0.50  1
163  39    Male  1.9  0.9  180   42   62  7.4  4.3  1.38  1
173  31    Male  0.6  0.1  175   48   34  6.0  3.7  1.60  1
200  49    Male  0.6  0.1  218   50   53  5.0  2.4  0.90  1

Duplikat Data Setelah Dihapus:
     65 Female   0.7  0.1  187  16   18  6.8  3.3   0.9  1
0    62   Male  10.9  5.5  699  64  10

# Soal No 6

In [ ]:
proses dimana dataset dikelompokkan lebih spesifik. Fungsi yang digunakan yaitu groupby(column_name). Misalnya ingin mengumpulkan data penjualan berdasarkan lokasi cabang toko (nama kolom : Cabang). Maka kita bisa menggabungkan data dengan fungsi groupby().

# Soal No 7

In [10]:
import pandas as pd
file_path = 'Uniqlo(FastRetailing) 2012-2016 Training - stocks2012-2016.csv'
pd.read_csv(file_path)

Date   Open   High    Low  Close   Volume  Stock Trading
0     2016-12-30  42120  42330  41700  41830   610000    25628028000
1     2016-12-29  43000  43220  42540  42660   448400    19188227000
2     2016-12-28  43940  43970  43270  43270   339900    14780670000
3     2016-12-27  43140  43700  43140  43620   400100    17427993000
4     2016-12-26  43310  43660  43090  43340   358200    15547803000
...          ...    ...    ...    ...    ...      ...            ...
1221  2012-01-11  14360  14750  14280  14590  1043400    15191988000
1222  2012-01-10  13890  14390  13860  14390   952300    13533413000
1223  2012-01-06  13990  14030  13790  13850   765500    10635609000
1224  2012-01-05  13720  13840  13600  13800   511500     7030811000
1225  2012-01-04  14050  14050  13700  13720   559100     7719804000

[1226 rows x 7 columns]

# Soal No 8

In [2]:
import pandas as pd
file_path = 'Uniqlo(FastRetailing) 2012-2016 Training - stocks2012-2016.csv'
df=pd.read_csv(file_path)
df.groupby("Open").first()


Date    Low  Close   Volume  Stock Trading
Open  High                                                   
13720 13840  2012-01-05  13600  13800   511500     7030811000
13890 14390  2012-01-10  13860  14390   952300    13533413000
13990 14030  2012-01-06  13790  13850   765500    10635609000
14050 14050  2012-01-04  13700  13720   559100     7719804000
14360 14750  2012-01-11  14280  14590  1043400    15191988000
...                 ...    ...    ...      ...            ...
59880 60460  2015-07-29  59480  60230   559600    33606522000
60450 61830  2015-08-04  60360  61380   502800    30855050000
60800 61970  2015-07-30  60740  61930   737700    45380183000
60820 61280  2015-08-03  60440  60740   363300    22071037000
61550 61650  2015-07-31  60670  61370   498200    30521965000

[1217 rows x 5 columns]

# Soal No 9

In [3]:
import pandas as pd

file_path = 'Uniqlo(FastRetailing) 2012-2016 Training - stocks2012-2016.csv'
df = pd.read_csv(file_path)
grouped_data = df.groupby("Open")
result = grouped_data[['High', 'Low']].mean()
print(result)

          High      Low
Open                   
13720  13840.0  13600.0
13890  14390.0  13860.0
13990  14030.0  13790.0
14050  14050.0  13700.0
14360  14750.0  14280.0
...        ...      ...
59880  60460.0  59480.0
60450  61830.0  60360.0
60800  61970.0  60740.0
60820  61280.0  60440.0
61550  61650.0  60670.0

[943 rows x 2 columns]


# Soal No 10

In [5]:
import pandas as pd
file_path = 'Uniqlo(FastRetailing) 2012-2016 Training - stocks2012-2016.csv'
df=pd.read_csv(file_path)
df.groupby(["Open","High"]).first()

Date    Low  Close   Volume  Stock Trading
Open  High                                                   
13720 13840  2012-01-05  13600  13800   511500     7030811000
13890 14390  2012-01-10  13860  14390   952300    13533413000
13990 14030  2012-01-06  13790  13850   765500    10635609000
14050 14050  2012-01-04  13700  13720   559100     7719804000
14360 14750  2012-01-11  14280  14590  1043400    15191988000
...                 ...    ...    ...      ...            ...
59880 60460  2015-07-29  59480  60230   559600    33606522000
60450 61830  2015-08-04  60360  61380   502800    30855050000
60800 61970  2015-07-30  60740  61930   737700    45380183000
60820 61280  2015-08-03  60440  60740   363300    22071037000
61550 61650  2015-07-31  60670  61370   498200    30521965000

[1217 rows x 5 columns]

# Soal No 11

In [7]:
import pandas as pd
file_path = 'Uniqlo(FastRetailing) 2012-2016 Training - stocks2012-2016.csv'
df = pd.read_csv(file_path)
grouped_data = df.groupby("High")
max_idx = grouped_data['Low'].idxmax()
min_idx = grouped_data['Low'].idxmin()
max_rows = df.loc[max_idx]
min_rows = df.loc[min_idx]
print("Baris dengan nilai High maksimal:")
print(max_rows)
print("\nBaris dengan nilai High minimal:")
print(min_rows)

Baris dengan nilai High maksimal:
            Date   Open   High    Low  Close  Volume  Stock Trading
1224  2012-01-05  13720  13840  13600  13800  511500     7030811000
1223  2012-01-06  13990  14030  13790  13850  765500    10635609000
1225  2012-01-04  14050  14050  13700  13720  559100     7719804000
1222  2012-01-10  13890  14390  13860  14390  952300    13533413000
1220  2012-01-12  14520  14600  14330  14520  540500     7814121000
...          ...    ...    ...    ...    ...     ...            ...
348   2015-07-29  59880  60460  59480  60230  559600    33606522000
345   2015-08-03  60820  61280  60440  60740  363300    22071037000
346   2015-07-31  61550  61650  60670  61370  498200    30521965000
344   2015-08-04  60450  61830  60360  61380  502800    30855050000
347   2015-07-30  60800  61970  60740  61930  737700    45380183000

[968 rows x 7 columns]

Baris dengan nilai High minimal:
            Date   Open   High    Low  Close  Volume  Stock Trading
1224  2012-01-05  13720 

# Soal No 12

In [8]:
import pandas as pd

# Meminta pengguna untuk memasukkan jumlah mahasiswa
num_students = int(input("Masukkan jumlah mahasiswa: "))

# Inisialisasi list untuk menyimpan data nilai mahasiswa
data = []

# Memasukkan nilai mahasiswa oleh pengguna
for i in range(num_students):
    nama_mahasiswa = input(f"Masukkan nama mahasiswa ke-{i+1}: ")
    nilai = float(input(f"Masukkan nilai {nama_mahasiswa}: "))
    data.append({'Nama Mahasiswa': nama_mahasiswa, 'Nilai': nilai})

# Membuat DataFrame dari list data
df = pd.DataFrame(data)

# Menyimpan DataFrame ke dalam file CSV
file_path = 'nilai_mahasiswa.csv'
df.to_csv(file_path, index=False)

print(f"Data nilai mahasiswa telah disimpan dalam file: {file_path}")

# Menemukan nilai maksimal dan minimal
nilai_terbesar = df.loc[df['Nilai'].idxmax()]
nilai_terkecil = df.loc[df['Nilai'].idxmin()]

# Menampilkan hasil
print("\nHasil:")
print(f"Nilai terbesar:\n{nilai_terbesar}")
print(f"\nNilai terkecil:\n{nilai_terkecil}")

Masukkan jumlah mahasiswa:  10
Masukkan nama mahasiswa ke-1:  a
Masukkan nilai a:  100
Masukkan nama mahasiswa ke-2:  b
Masukkan nilai b:  90
Masukkan nama mahasiswa ke-3:  c
Masukkan nilai c:  80
Masukkan nama mahasiswa ke-4:  d
Masukkan nilai d:  70
Masukkan nama mahasiswa ke-5:  e
Masukkan nilai e:  60
Masukkan nama mahasiswa ke-6:  f
Masukkan nilai f:  50
Masukkan nama mahasiswa ke-7:  g
Masukkan nilai g:  40
Masukkan nama mahasiswa ke-8:  h
Masukkan nilai h:  30
Masukkan nama mahasiswa ke-9:  i
Masukkan nilai i:  20
Masukkan nama mahasiswa ke-10:  j
Masukkan nilai j:  10


Data nilai mahasiswa telah disimpan dalam file: nilai_mahasiswa.csv

Hasil:
Nilai terbesar:
Nama Mahasiswa        a
Nilai             100.0
Name: 0, dtype: object

Nilai terkecil:
Nama Mahasiswa       j
Nilai             10.0
Name: 9, dtype: object


# Soal No 13

# Soal No 14

Berikut ini adalah data nilai mahasiswa
Andi     70
Rachel   80
Udin     35

Buatlah program untuk mencari nilai mahasiswa dengan ketentuan :
1. Data Mahasiswa Dimasukan Dari User
2. Data Nilai yang akan dicari dimasukan dari user
3. Tampilkan data nilai

In [ ]:
# Membuat dictionary untuk menyimpan data mahasiswa dan nilai
data_mahasiswa = {}

# Meminta pengguna untuk memasukkan data mahasiswa dan nilai
while True:
    nama = input("Masukkan nama mahasiswa (atau ketik 'selesai' untuk selesai): ")
    
    if nama.lower() == 'selesai':
        break
    
    try:
        nilai = float(input(f"Masukkan nilai {nama}: "))
        data_mahasiswa[nama] = nilai
    except ValueError:
        print("Masukkan nilai yang valid.")

# Menampilkan data nilai mahasiswa
print("\nData Nilai Mahasiswa:")
for nama, nilai in data_mahasiswa.items():
    print(f"{nama}: {nilai}")

# Meminta pengguna untuk memasukkan nilai yang ingin dicari
try:
    nilai_cari = float(input("Masukkan nilai yang ingin dicari: "))
except ValueError:
    print("Masukkan nilai yang valid.")
    exit()

# Mencari dan menampilkan data mahasiswa berdasarkan nilai yang dimasukkan pengguna
mahasiswa_terpilih = [(nama, nilai) for nama, nilai in data_mahasiswa.items() if nilai == nilai_cari]

if mahasiswa_terpilih:
    print(f"\nData mahasiswa dengan nilai {nilai_cari}:")
    for nama, nilai in mahasiswa_terpilih:
        print(f"{nama}: {nilai}")
else:
    print(f"Tidak ada mahasiswa dengan nilai {nilai_cari}")